pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -90,6 +99,12 @@ func NewClusterMemberRemovalController(
 }
 
 func (c *clusterMemberRemovalController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	// skip reconciling this controller, if it is unsafe to scale the etcd cluster membership
+	err := ceohelpers.CheckSafeToScaleCluster(c.configMapLister, c.operatorClient, c.namespaceLister, c.infrastructureLister, c.etcdClient)
+	if err != nil {
+		return err
+	}
```

Comment:
that's already your job done, you don't need to add any other code change in this controller. At this point, there is only a safe cluster without a bootstrap node and etcd member.

Reply:
apologies, I got the conditions backwards. At this point there may be a bootstrap member still present. So instead of this `CheckSafeToScaleCluster` you can use `bootstrapComplete, err := IsBootstrapComplete(configmapLister, staticPodClient, etcdClient)`.

Only continue the sync loop if bootstrap is really completed.

Reply:
done 👍🏽

Reply:
just one last nitpick:

> 		klog.V(2).Infof("bootstrapping is not complete, ignoring scaling down")

this can get very spammy, especially since it's syncing every minute or so with the node status.
I would just leave this log out.